app/src/main/java/org/wikipedia/login/LoginActivity.kt

```diff
@@ -0,0 +1,265 @@
+package org.wikipedia.login
+
+import android.accounts.AccountAuthenticatorResponse
+import android.accounts.AccountManager
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import com.google.android.material.textfield.TextInputLayout
+import org.apache.commons.lang3.StringUtils
+import org.wikipedia.Constants
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.activity.BaseActivity
+import org.wikipedia.analytics.LoginFunnel
+import org.wikipedia.auth.AccountUtil.updateAccount
+import org.wikipedia.createaccount.CreateAccountActivity
+import org.wikipedia.databinding.ActivityLoginBinding
+import org.wikipedia.login.LoginClient.LoginFailedException
+import org.wikipedia.notifications.NotificationPollBroadcastReceiver.Companion.pollNotifications
+import org.wikipedia.page.PageTitle
+import org.wikipedia.push.WikipediaFirebaseMessagingService.Companion.updateSubscription
+import org.wikipedia.readinglist.sync.ReadingListSyncAdapter
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.DeviceUtil.hideSoftKeyboard
+import org.wikipedia.util.FeedbackUtil.showError
+import org.wikipedia.util.FeedbackUtil.showMessage
+import org.wikipedia.util.FeedbackUtil.showPrivacyPolicy
+import org.wikipedia.util.UriUtil.visitInExternalBrowser
+import org.wikipedia.util.log.L.w
```

Comment:
We prefer to use qualified static methods in the code itself (e.g. `L.w()`), instead of static imports (e.g. `w()`).